### PR TITLE
[INTERNAL] SourceMaps follow-up

### PR DESCRIPTION
**Pull request by matz3**
Follow-up of https://github.com/SAP/ui5-builder/pull/695

### CODE REVIEW

File: lib/processors/minifier.js [hunk 1]
```diff
@@ -34,12 +34,16 @@ const debugFileRegex = /((?:\.view|\.fragment|\.controller|\.designtime|\.suppor
...
- * @param {boolean} [parameters.addSourceMappingUrl=true]
+ * @param {object} [parameters.options] Options
+ * @param {boolean} [parameters.options.addSourceMappingUrl=true]
  * 				Whether to add a sourceMappingURL reference to the end of the minified resource
  * @returns {Promise<module:@ui5/builder.processors.MinifierResult[]>}
  * 				Promise resolving with object of resource, dbgResource and sourceMap
  */
-module.exports = async function({resources, addSourceMappingUrl = true}) {
+module.exports = async function({resources, options: {addSourceMappingUrl} = {}}) {
+	if (addSourceMappingUrl === undefined) {
```
Comment (flovogt) [inline lib/processors/minifier.js:44]: is it possible to specify the default value also for nested parameters in the function declaration?

File: test/lib/processors/minifier.js [hunk 2]
```diff
@@ -200,10 +206,84 @@ test();`;
...
 	const expected = `//@ui5-bundle sap/ui/my/module.js
 function test(t){var o=t;console.log(o)}test();
-//# sourceMappingURL=test.js.map`;
+${SOURCE_MAPPING_URL}=test.js.map`;
+	t.deepEqual(await resource.getString(), expected, "Correct minified content");
+});
+
+test("addSourceMappingUrl=false", async (t) => {
+	const content = `
+//@ui5-bundle sap/ui/my/module.js
```
Comment (flovogt) [inline test/lib/processors/minifier.js:215]: should we add there some indentation?

File: test/lib/processors/minifier.js [hunk 3]
```diff
@@ -200,10 +206,84 @@ test();`;
...
+	});
+	const [{resource}] = await minifier({
+		resources: [testResource],
+		options: {
+			addSourceMappingUrl: false
+		}
+	});
+
+	const expected = `//@ui5-bundle sap/ui/my/module.js
+function test(t){var o=t;console.log(o)}test();`;
```
Comment (flovogt) [inline test/lib/processors/minifier.js:234]: should we add there some indentation?

File: test/lib/processors/minifier.js [hunk 4]
```diff
@@ -200,10 +206,84 @@ test();`;
...
+	t.deepEqual(await resource.getString(), expected, "Correct minified content");
+});
+
+test("addSourceMappingUrl=true", async (t) => {
+	const content = `
+//@ui5-bundle sap/ui/my/module.js
```
Comment (flovogt) [inline test/lib/processors/minifier.js:240]: should we add there some indentation?

File: test/lib/processors/minifier.js [hunk 5]
```diff
@@ -200,10 +206,84 @@ test();`;
...
+
+	const expected = `//@ui5-bundle sap/ui/my/module.js
+function test(t){var o=t;console.log(o)}test();
+${SOURCE_MAPPING_URL}=test.js.map`;
 	t.deepEqual(await resource.getString(), expected, "Correct minified content");
 });
 
+test("empty options object (addSourceMappingUrl defaults to true)", async (t) => {
+	const content = `
+//@ui5-bundle sap/ui/my/module.js
```
Comment (flovogt) [inline test/lib/processors/minifier.js:266]: should we add there some indentation?

File: test/lib/tasks/minify.js [hunk 6]
```diff
@@ -148,3 +152,94 @@ test();`;
...
+	if (!resSourceMap) {
+		t.fail("Could not find /test-dbg.js.map in target locator");
+	}
+	t.deepEqual(await resSourceMap.getString(), expectedSourceMap, "Correct source map content");
+});
+
+test("integration: minify omitSourceMapResources=false (without taskUtil)", async (t) => {
+	const {reader, writer, workspace} = createWorkspace();
+	const content = `
+function test(paramA) {
```
Comment (flovogt) [inline test/lib/tasks/minify.js:203]: should we add there some indentation?

File: test/lib/tasks/minify.js [hunk 7]
```diff
@@ -148,3 +152,94 @@ test();`;
 		"Third taskUtil.setTag call with expected arguments");
 });
 
+test("integration: minify (without taskUtil)", async (t) => {
+	const {reader, writer, workspace} = createWorkspace();
+	const content = `
+function test(paramA) {
```
Comment (flovogt) [inline test/lib/tasks/minify.js:158]: should we add there some indentation?

**Review by flovogt — COMMENTED** [review]
Only minor things. Might be non relevant

File: lib/processors/minifier.js (hunk 1 above)
Comment (matz3) [reply lib/processors/minifier.js:44]: Thanks. I thought that this wouldn't work when no options are provided. But it seems to work.

Comment (matz3) [reply lib/processors/minifier.js]: Good point, thanks. I thought that it isn't possible when the options should be optional, but it seems to work.

File: test/lib/processors/minifier.js (hunk 2 above)
Comment (matz3) [reply test/lib/processors/minifier.js:215]: This is by intention as the file content should not have any leading tabs or spaces.

Comment (flovogt) [reply test/lib/processors/minifier.js:215]: Thx

File: test/lib/tasks/minify.js (hunk 6 above)
Comment (flovogt) [reply test/lib/tasks/minify.js:203]: answered above https://github.com/SAP/ui5-builder/pull/709#discussion_r823602439

File: test/lib/tasks/minify.js (hunk 7 above)
Comment (flovogt) [reply test/lib/tasks/minify.js:158]: answered above https://github.com/SAP/ui5-builder/pull/709#discussion_r823602439

File: test/lib/processors/minifier.js (hunk 5 above)
Comment (flovogt) [reply test/lib/processors/minifier.js:266]: answered above https://github.com/SAP/ui5-builder/pull/709#discussion_r823602439

File: lib/processors/minifier.js (hunk 1 above)
Comment (flovogt) [reply lib/processors/minifier.js]: Fixed by https://github.com/SAP/ui5-builder/pull/709/commits/267fafc0da8ed36f859423cb1e2010b84c9fb11a

**Review by flovogt — APPROVED** [review]
LGTM

**Pull request merged**